### PR TITLE
Only support code coverage tool for VM apps 

### PR DESCRIPTION
**Pull request by elliette**
Fixes https://github.com/flutter/devtools/issues/4784

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/debugger/controls.dart [hunk 1]
```diff
@@ -54,7 +56,7 @@ class _DebuggingControlsState extends State<DebuggingControls>
           const SizedBox(width: denseSpacing),
           BreakOnExceptionsControl(controller: controller),
           const SizedBox(width: denseSpacing),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/debugger/controls.dart:58]: group this SizedBox with the CodeCoverageToggle 
```
if (isVmAp) ...[
 ...
],
```

File: packages/devtools_app/test/debugger/debugger_codeview_test.dart [hunk 2]
```diff
@@ -45,6 +45,7 @@ void main() {
 
     when(fakeServiceManager.connectedApp!.isProfileBuildNow).thenReturn(false);
     when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
+    when(fakeServiceManager.connectedApp!.isRunningOnDartVM).thenReturn(true);
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/debugger/debugger_codeview_test.dart:48]: use the mockConnectedApp helper

**Review by kenzieschmoll — APPROVED** [review]
couple comments then lgtm

**Review by bkonyi — APPROVED** [review]
LGTM once Kenzie's comments are addressed.

File: packages/devtools_app/test/debugger/debugger_codeview_test.dart (hunk 2 above)
Comment (elliette) [reply packages/devtools_app/test/debugger/debugger_codeview_test.dart]: done!

File: packages/devtools_app/lib/src/screens/debugger/controls.dart (hunk 1 above)
Comment (elliette) [reply packages/devtools_app/lib/src/screens/debugger/controls.dart]: done!

**Pull request merged**